pyDeltaRCM/init_tools.py

```diff
@@ -449,40 +474,61 @@ def init_output_grids(self):
             y[:] = self.y
 
+            # set up variables for output data grids
             if self.save_eta_grids:
-                eta = self.output_netcdf.createVariable('eta',
-                                                        'f4',
-                                                        ('total_time', 'length', 'width'))
+                eta = self.output_netcdf.createVariable(
+                    'eta', 'f4', ('total_time', 'length', 'width'))
                 eta.units = 'meters'
-
             if self.save_stage_grids:
-                stage = self.output_netcdf.createVariable('stage',
-                                                          'f4',
-                                                          ('total_time', 'length', 'width'))
+                stage = self.output_netcdf.createVariable(
+                    'stage', 'f4', ('total_time', 'length', 'width'))
                 stage.units = 'meters'
-
             if self.save_depth_grids:
-                depth = self.output_netcdf.createVariable('depth',
-                                                          'f4',
-                                                          ('total_time', 'length', 'width'))
+                depth = self.output_netcdf.createVariable(
+                    'depth', 'f4', ('total_time', 'length', 'width'))
                 depth.units = 'meters'
-
             if self.save_discharge_grids:
-                discharge = self.output_netcdf.createVariable('discharge',
-                                                              'f4',
-                                                              ('total_time', 'length', 'width'))
+                discharge = self.output_netcdf.createVariable(
+                    'discharge', 'f4', ('total_time', 'length', 'width'))
                 discharge.units = 'cubic meters per second'
-
             if self.save_velocity_grids:
-                velocity = self.output_netcdf.createVariable('velocity',
-                                                             'f4',
-                                                             ('total_time', 'length', 'width'))
+                velocity = self.output_netcdf.createVariable(
+                    'velocity', 'f4', ('total_time', 'length', 'width'))
                 velocity.units = 'meters per second'
+            if self.save_sedflux_grids:
+                sedflux = self.output_netcdf.createVariable(
+                    'sedflux', 'f4', ('total_time', 'length', 'width'))
+                sedflux.units = 'cubic meters per second'
+
+            # set up metadata group and populate variables
+            def _create_meta_variable(varname, varvalue, varunits,
+                                      vartype='f4', vardims=()):
+                _v = self.output_netcdf.createVariable(
+                    'meta/'+varname, vartype, vardims)
+                _v.units = varunits
+                _v[:] = varvalue
+
+            if self.save_metadata:
```

Comment:
Maybe a controversial opinion here, but I'd be in favor of removing this `if` clause. Then if any variable requires the netCDF to be created, the metadata would get created and saved.

I'm hard-pressed to imagine the small bits of memory the metadata occupies would bother someone who didn't necessarily want it to be included. But it is easy for me to imagine someone setting up one or many runs with the intention of saving the metadata but missing that `bool` in the yaml, and ending up with a bunch of `.nc` files with grids and no associated metadata.

Reply:
hmm nah, I think you're right. so jsut so I'm clear:

1) if any `grids` are saved, also save the metadata
2) if no grids are saves, check whether `save_metadata` is true

Reply:
Yes, I think you could accomplish that by just dropping the if statement from line 515 and removing one indent for the lines below that, since lines 430-437 check the grids/metadata to see if the netCDF should even be created.

Reply:
okay, I got this implemented and tested.